lib/clAdapters/activeSync/calendar/index.ts

```diff
@@ -0,0 +1,310 @@
+  'organizerEmail':                      'OrganizerEmail',
+  'recurrence':                          'Recurrence',
+  'responseRequested':                   'ResponseRequested',
+  'responseStatus':                      'ResponseType',
+  // 'seriesMasterId':                      'recurringEventId', // Does not appear to be available
+  'dateTimeStart':                       'StartTime',
+  'dateTimeEnd':                         'EndTime',
+  'subject':                             'Subject',
+  'url':                                 'WebLink',
+  // 'hangoutLink':                         'hangoutLink',  // Does not appear to be available
+  'privacy':                             'Sensitivity'
+};
+
+
+export default class ActiveSyncCalendarAdapter extends ActiveSyncBaseAdapter {
+
+  static Configuration = Configuration;
+  static Service = Service;
+
+  // convert the names of the api response data
+  static fieldNameMap = fieldNameMap;
+
+
+
+  _config: Configuration;
+  _service: Service;
+
+  // constructor needs to call super
+  constructor() {
+    super();
+  }
+
+
+  reset() {
+    delete this._config;
+    delete this._service;
+    return this;
+  }
+
+
+  async init() {
+
+    const { credentials }: { credentials: {[k: string]: string} } = this;
+
+    if (!credentials) {
+      throw new Error('credentials required for adapter.');
+    }
+
+    // map Google json keys to keys used in this library
+    for (const want in credentialMappings) {
+      const alternate = credentialMappings[want];
+      if (!credentials[want]) {
+        credentials[want] = credentials[alternate];
+      }
+    }
+
+    // validate required credential properties
+    Object.keys(credentialMappings)
+      .forEach(prop => {
+        if (!credentials[prop]) {
+          throw new Error(`Property ${prop} required in adapter credentials!`);
+        }
+      });
+
+    this._config  = new ActiveSyncCalendarAdapter.Configuration(credentials);
+    this._service = new ActiveSyncCalendarAdapter.Service(this._config);
+
+    await this._service.init();
+
+    const { email: email } = credentials;
+
+    console.log(
+      `Successfully initialized active sync calendar adapter for email: ${email}`
+    );
+
+    return this;
+  }
+
+  async getData(filterStartDate: Date, filterEndDate: Date, properties: any) {
+    if (filterStartDate.getTime() > filterEndDate.getTime()) {
+      throw new Error(`filterStartDate must be less than or equal to filterEndDate`);
+    }
+
+    const { email: username, password, connectUrl: endpoint } = this.credentials;
+
+    const individualRunStats = {
+      filterStartDate,
+      filterEndDate,
+      success: true,
+      runDate: moment().utc().toDate(),
+      errorMessage: null
+    };
+
+    try {
+      const options = {
+        username,
+        password,
+        endpoint,
+        folders : [],
+        folderSyncKey : '0',
+        device: {
+          id: '0000000',
```

Comment:
According to https://technet.microsoft.com/en-us/library/exchange-online-limits.aspx, 100 is the "maximum number of Exchange ActiveSync devices per mailbox."  It seems we would be safe from hitting that limit with the device id of '000000'.